dcicutils/deployment_utils.py

```diff
@@ -227,6 +227,29 @@ def extract_env_name_configuration(cls, client, env_name):
         configurable_options = env_settings[EB_OPTION_SETTINGS]
         return configurable_options
 
+    @classmethod
+    def verify_template_creation(cls, client, template_name):
+        """ Does a get for the given template_name to verify EB has recognized that it is
+            available.
+
+        :param client: boto3 elasticbeanstalk client
+        :param template_name: name of template to check
+        :returns: True if template can be acquired
+        :raises: Exception if one is encountered
+        """
+        try:
```

Comment:
maybe this (my actual recommendation of these three):
```
    for retrying in (False, True):
        try:
            client.describe_configuration_settings(
                ApplicationName=cls.EB_APPLICATION,
                TemplateName=template_name
            )
        except Exception:
            if retrying:
                raise
            else:
                time.sleep(10)
    return True
```
Possibly also this, though it's more obscure:
```
    for attempt in range(2):
        if attempt != 0: time.sleep(10)
        try:
            client.describe_configuration_settings(
                ApplicationName=cls.EB_APPLICATION,
                TemplateName=template_name
            )
            return True
        except Exception:
            pass
    raise
```
and probably no one ever uses `else` in `for` but you can here and it's awful:
```
    for attempt in range(2):
        if attempt != 0: time.sleep(10)
        try:
            client.describe_configuration_settings(
                ApplicationName=cls.EB_APPLICATION,
                TemplateName=template_name
            )
            break
        except Exception:
            pass
    else:
        raise
    return True
```

Reply:
Im taking the first one but flipping the retry order, feels more clear to me:
```
        for retrying in (True, False):
            try:
                client.describe_configuration_settings(
                    ApplicationName=cls.EB_APPLICATION,
                    TemplateName=template_name
                )
            except Exception:
                if not retrying:
                    raise
                else:
                    time.sleep(10)
        return True
```